core/templates/dev/head/components/fuzzy_rule_editor_directive.html

```diff
@@ -0,0 +1,2 @@
+<input type="hidden"/>
+<button type="button" class="btn btn-danger" ng-click="removeEntry()">Remove this input</button>
```

Comment:
Thanks for information. I did not know that we already had `<select2-dropdown>`. Anyway I fixed the `ng-click` not triggerring issue by using this directive. But another issue arose in changing `choices / options` of `select2-dropdown` dynamically. So that choices are updated whenever some training answer is removed using `Remove this input` button. It turns out that a lot of people have this [issue](https://github.com/select2/select2/issues/2830) and since there is no clean way to do it in select2, many people have derived their own hacks to do it. I used one of the hacks from that issue to fix it.